File: web/src/service/article.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Article} from '../entity/article';
import {Page} from '@yunzhi/ng-common';

@Injectable({
  providedIn: 'root'
})
export class ArticleService {

  constructor(private httpClient: HttpClient) {
  }

  getTop10Articles(): Observable<Article[]> {
    return this.httpClient.get<Article[]>('/article/getTop10Articles')
  }

  all(): Observable<Article[]> {
    return this.httpClient.get<Article[]>('/article/getAll');
  }

  page(param: { size: number; page: number; title: string }) {
    return this.httpClient.get<Page<Article>>('/article/page', {params: param})
  }

  save(article: { title: string, content: string, description: string }): Observable<Article> {
    article = article as Article;
    return this.httpClient.post<Article>('/article', article);
  }

  getById(id: number): Observable<Article> {
    return this.httpClient.get<Article>(`/article/${id}`);
  }

  update(id: number, article: Article) {
    return this.httpClient.put(`/article/${id}`, article);
  }
}

File: api/src/main/java/com/kxb/employmentServices/service/ArticleService.java
package com.kxb.employmentServices.service;

import com.kxb.employmentServices.dto.ArticleDto;
import com.kxb.employmentServices.entity.Article;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ArticleService {

    Article getById(Long id);

    List<Article> getTop10Articles();

    Page<Article> page(Pageable pageable, String title);

    Article save(ArticleDto.SaveRequest request);

    Article update(Long id, ArticleDto.UpdateRequest request);

    List<Article> getAll();

    Article updateHits(Long id);
}
